### DISCUSSION TITLE
Migrating conbee ll -> sonoff-p: Edited: Z2M not starting, comissioning issue?

### DISCUSSION THREAD

**Discussion by ThorAlex87** [Koenkk/zigbee2mqtt#18871] [General]
Hi! So I'm trying to migrate home assistant (HA OS on rpi4, Z2M addon) from a conbee ll to a sonoff-p, following the instructions here: https://www.zigbee2mqtt.io/guide/faq/#how-do-i-migrate-from-one-adapter-to-another

> First question, ignore as i'm a dummy:

> Now after completing the procedure Z2M will not start, and in the log i find:
> 
> > Zigbee2MQTT:error 2023-09-05 17:29:26: Error: Error while opening serialport 'Error: Error: No such file or directory, cannot open /dev/ttyACM0'
> >     at SerialPort.<anonymous> (/app/node_modules/zigbee-herdsman/src/adapter/deconz/driver/driver.ts:200:28)
> >     at SerialPort._error (/app/node_modules/@serialport/stream/dist/index.js:75:22)
> >     at /app/node_modules/@serialport/stream/dist/index.js:111:18
> 
>     
> But, my configuration.yaml is updated to:
> 
> > serial:
> >   port: /dev/ttyUSB0
> >   adapter: ITead
> 
>   
> that should be correct for the new adapter according to hardware info.
> 
> 
> Looks like Z2M is still trying to connect to the old adapter and disregarding the updates to the config file? I've of course rebooted everything several times, with no change. 

Edit 1: So i fixed my first problem, by configuring the addon rather than editing configuration.yaml. Now I have a new issue, what is the correct configuration for "adapter:"?

If i use "adapter: ezsp" I get:
> Error: Reset error: Error: {"sequence":-1} after 10000ms
    at /app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/uart.ts:299:23
    at Queue.executeNext (/app/node_modules/zigbee-herdsman/src/utils/queue.ts:32:32)
    
If I use "adapter: zstack" I get:
> Zigbee2MQTT:error 2023-09-05 18:41:15: Error: network commissioning timed out - most likely network with the same panId or extendedPanId already exists nearby
    at ZnpAdapterManager.beginCommissioning (/app/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/manager.ts:347:23)
    at ZnpAdapterManager.start (/app/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/manager.ts:92:17)
    at Controller.start (/app/node_modules/zigbee-herdsman/src/controller/controller.ts:132:29)
    at Zigbee.start (/app/lib/zigbee.ts:60:27)
    at Controller.start (/app/lib/controller.ts:101:27)
    at start (/app/index.js:107:5)

If i use "adapter: zigate" I get:
> error 2023-09-05 18:43:49: Error: failed to connect to zigate adapter [object Object] after 5000ms
    at ZiGateAdapter.start (/app/node_modules/zigbee-herdsman/src/adapter/zigate/adapter/zigateAdapter.ts:101:19)
    at runNextTicks (node:internal/process/task_queues:61:5)
    at listOnTimeout (node:internal/timers:528:9)
    at processTimers (node:internal/timers:502:7)
    at Controller.start (/app/node_modules/zigbee-herdsman/src/controller/controller.ts:132:29)
    at Zigbee.start (/app/lib/zigbee.ts:60:27)
    at Controller.start (/app/lib/controller.ts:101:27)
    at start (/app/index.js:107:5)
  
Despite googlein I can finde no clear answer to what it should be, and thess errors are not very helpful to me... Maybe zstack is more promesing as it's not a timeoul, but i have no idea what it means.

Edit 2: Concluded zstack was the right answer.  Found a thread (https://github.com/Koenkk/zigbee2mqtt/issues/10209) of other people getting the same error, tried different usb cable and flashing newest firmware but still get the exact same error. Anyone know what this panId stuff is, i can find no trace of it in my config file but from reading around it should be there? Anywas, I've reverted to the conbee for now as it's time for bed, will try again tomorrow.

Anyone have any ideas what is happening here and how to fix it?

Thanks!

**Comment by francisp2** [comment]
see: 
https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html#error-coordinator-failed-to-start-probably-the-panid-is-already-in-use-try-a-different-panid-or-channel

**Comment by Hedda** [comment]
> I'm trying to migrate home assistant (HA OS on rpi4, Z2M addon) from a conbee ll to a sonoff-p, following the instructions here: https://www.zigbee2mqtt.io/guide/faq/#how-do-i-migrate-from-one-adapter-to-another

See existing discussion -> https://github.com/Koenkk/zigbee2mqtt/discussions/16478
